dpnp/backend/extensions/lapack/evd_batch_common.hpp

```diff
@@ -0,0 +1,138 @@
+
+namespace dpctl_td_ns = dpctl::tensor::type_dispatch;
+namespace py = pybind11;
+
+template <typename dispatchT>
+std::pair<sycl::event, sycl::event>
+    evd_batch_func(sycl::queue &exec_q,
+                   const std::int8_t jobz,
+                   const std::int8_t upper_lower,
+                   dpctl::tensor::usm_ndarray &eig_vecs,
+                   dpctl::tensor::usm_ndarray &eig_vals,
+                   const std::vector<sycl::event> &depends,
+                   const dispatchT &evd_batch_dispatch_table)
+{
+    const int eig_vecs_nd = eig_vecs.get_ndim();
+
+    const py::ssize_t *eig_vecs_shape = eig_vecs.get_shape_raw();
+    const py::ssize_t *eig_vals_shape = eig_vals.get_shape_raw();
+
+    const int expected_eig_vecs_nd = 3;
+    const int expected_eig_vals_nd = 2;
+
+    common_evd_checks(exec_q, eig_vecs, eig_vals, eig_vecs_shape,
+                      expected_eig_vecs_nd, expected_eig_vals_nd);
+
+    if (eig_vecs_shape[2] != eig_vals_shape[0] ||
+        eig_vecs_shape[0] != eig_vals_shape[1])
+    {
+        throw py::value_error(
+            "The shape of 'eig_vals' must be (batch_size, n), "
+            "where batch_size = " +
+            std::to_string(eig_vecs_shape[0]) +
+            " and n = " + std::to_string(eig_vecs_shape[1]));
+    }
+
+    size_t src_nelems(1);
+
+    for (int i = 0; i < eig_vecs_nd; ++i) {
+        src_nelems *= static_cast<size_t>(eig_vecs_shape[i]);
+    }
+
+    if (src_nelems == 0) {
+        // nothing to do
+        return std::make_pair(sycl::event(), sycl::event());
+    }
+
+    auto array_types = dpctl_td_ns::usm_ndarray_types();
+    const int eig_vecs_type_id =
+        array_types.typenum_to_lookup_id(eig_vecs.get_typenum());
+    const int eig_vals_type_id =
+        array_types.typenum_to_lookup_id(eig_vals.get_typenum());
+
+    evd_batch_impl_fn_ptr_t evd_batch_fn =
+        evd_batch_dispatch_table[eig_vecs_type_id][eig_vals_type_id];
+    if (evd_batch_fn == nullptr) {
+        throw py::value_error(
+            "Types of input vectors and result array are mismatched.");
+    }
+
+    char *eig_vecs_data = eig_vecs.get_data();
+    char *eig_vals_data = eig_vals.get_data();
+
+    const std::int64_t batch_size = eig_vecs_shape[2];
+    const std::int64_t n = eig_vecs_shape[1];
+
+    const oneapi::mkl::job jobz_val = static_cast<oneapi::mkl::job>(jobz);
+    const oneapi::mkl::uplo uplo_val =
+        static_cast<oneapi::mkl::uplo>(upper_lower);
+
+    sycl::event evd_batch_ev =
+        evd_batch_fn(exec_q, jobz_val, uplo_val, batch_size, n, eig_vecs_data,
+                     eig_vals_data, depends);
+
+    sycl::event ht_ev = dpctl::utils::keep_args_alive(
+        exec_q, {eig_vecs, eig_vals}, {evd_batch_ev});
+
+    return std::make_pair(ht_ev, evd_batch_ev);
+}
+
+template <typename dispatchT,
+          template <typename fnT, typename T, typename RealT>
+          typename factoryT>
+void init_evd_batch_dispatch_table(
```

Comment:
The init functions in batch and non-batch implementations looks exactly the same.

Reply:
Done
Thanks!